modules/stock/commerce_pos_stock.module

```diff
@@ -41,3 +41,40 @@ function commerce_pos_stock_form_commerce_pos_return_alter(&$form, &$form_state,
     }
   }
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function commerce_pos_stock_form_commerce_pos_exchange_alter(&$form, &$form_state, $form_id) {
```

Comment:
There is already a form_ID alter for the return that adds the checkbox. I think using form_ID alter is a lot more performant than using a form alter which triggers every time just to add the checkbox for the returns and exchanges.